code/go/0chain.net/smartcontract/stakepool/spenum/enums.go

```diff
@@ -32,5 +32,12 @@ const (
 var poolString = []string{"active", "pending", "inactive", "unstaking", "deleting", "deleted"}
 
 func (p PoolStatus) String() string {
-	return poolString[p]
+	if int(p) < len(poolString) {
+		return poolString[p]
+	}
+	return ""
```

Comment:
Return value from poolString only when p is in [0, len(poolString)), so please also check if p is < 0. 

Return `unknown pool status` if p is invalid.